be1-go/hub/standard_hub/mod_test.go

```diff
@@ -303,20 +288,240 @@ func Test_Create_LAO(t *testing.T) {
 	name := "LAO X"
 
 	// LaoID is Hash(organizer||create||name) encoded in base64URL
-	h := sha256.New()
-	h.Write(keypair.publicBuf)
-	h.Write([]byte(fmt.Sprintf("%d", now)))
-	h.Write([]byte(name))
+	laoID := messagedata.Hash(base64.URLEncoding.EncodeToString(keypair.publicBuf), fmt.Sprintf("%d", now), name)
+
+	data := messagedata.LaoCreate{
+		Object:    messagedata.LAOObject,
+		Action:    messagedata.LAOActionCreate,
+		ID:        laoID,
+		Name:      name,
+		Creation:  now,
+		Organizer: base64.URLEncoding.EncodeToString(keypair.publicBuf),
+		Witnesses: []string{},
+	}
+
+	dataBuf, err := json.Marshal(data)
+	require.NoError(t, err)
+
+	signature, err := schnorr.Sign(suite, keypair.private, dataBuf)
+	require.NoError(t, err)
+
+	dataBase64 := base64.URLEncoding.EncodeToString(dataBuf)
+	expectedSignature := base64.URLEncoding.EncodeToString(signature)
 
-	laoID := base64.URLEncoding.EncodeToString(h.Sum(nil))
+	msg := message.Message{
+		Data:              "ThisIsNotBase64Encoded",
+		Sender:            base64.URLEncoding.EncodeToString(keypair.publicBuf),
+		Signature:         expectedSignature,
+		MessageID:         messagedata.Hash(dataBase64, expectedSignature),
+		WitnessSignatures: []message.WitnessSignature{},
+	}
+
+	publish := method.Publish{
+		Base: query.Base{
+			JSONRPCBase: jsonrpc.JSONRPCBase{
+				JSONRPC: "2.0",
+			},
+
+			Method: query.MethodPublish,
+		},
+
+		ID: 1,
+
+		Params: struct {
+			Channel string          `json:"channel"`
+			Message message.Message `json:"message"`
+		}{
+			Channel: "/root",
+			Message: msg,
+		},
+	}
+
+	publishBuf, err := json.Marshal(&publish)
+	require.NoError(t, err)
+
+	sock := &fakeSocket{}
+
+	hub.handleMessageFromClient(&socket.IncomingMessage{
+		Socket:  sock,
+		Message: publishBuf,
+	})
+
+	require.EqualError(t, sock.err, fmt.Sprintf("%v", sock.err))
+}
+
+func Test_Create_Invalid_Json_Schema(t *testing.T) {
```

Comment:
json schemas are tested in the validation folder

Reply:
Thank you for noting